plugin/src/main/java/org/opennms/velocloud/rest/v1/VelocloudRestServiceV1Impl.java

```diff
@@ -52,4 +58,69 @@ public List<EnterpriseDTO> getCustomersForMspPartner(final String alias) throws
                      .map(customer -> Mapper.ENTERPRISE_INSTANCE.sourceToTarget(customer))
                      .collect(Collectors.toList());
     }
+
+    @Override
+    public List<ConnectionListElementDTO> getConnectionList() {
+        return this.connectionManager.getAliases().stream()
+                .map(alias -> {
+                    var connection = connectionManager.getConnection(alias).orElseThrow();
+                    final var connectionDTO = new ConnectionListElementDTO();
+                    connectionDTO.setAlias(alias);
+                    connectionDTO.setOrchestratorUrl(connection.getOrchestratorUrl());
+                    return connectionDTO;
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Response addConnection(ConnectionDTO connectionDTO, boolean dryRun, boolean force) {
+        if (this.connectionManager.contains(connectionDTO.getAlias())) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("Connection already exists").build();
+        }
+        if (!force) {
+            final var exception = Optional.ofNullable(connectionManager.testConnection(connectionDTO.getOrchestratorUrl(), connectionDTO.getApiKey()));
+            if (exception.isPresent()) {
+                return Response.status(Response.Status.BAD_REQUEST).entity(String.format("Failed to validate credentials: %s", exception.get().getMessage())).build();
+            }
+        }
+        if (!dryRun) {
+            connectionManager.addConnection(connectionDTO.getAlias(), connectionDTO.getOrchestratorUrl(), connectionDTO.getApiKey());
+            return Response.ok().entity("Connection successfully added").build();
+        }
+        return Response.ok().entity("Dry run successful").build();
+    }
+
+    @Override
+    public Response editConnection(final String alias, final ConnectionDTO connectionDTO, boolean force) {
+        if (!this.connectionManager.contains(alias)) {
```

Comment:
I think it would be more idiomatic to rely on `ConnectionMananger.getConnection` here and use the tools `Optional` provides.